src/posix/platform/resolver.cpp

```diff
@@ -95,16 +97,26 @@ void Resolver::LoadDnsServerListFromConf(void)
 
     while (fp.good() && std::getline(fp, line) && mUpstreamDnsServerCount < kMaxUpstreamServerCount)
     {
-        if (line.find(kNameserverItem, 0) == 0)
+        // Skip the lines that don't start with "nameserver"
+        if (line.find(kNameserverItem, 0))
         {
-            in_addr_t addr;
+            continue;
+        }
 
-            if (inet_pton(AF_INET, &line.c_str()[sizeof(kNameserverItem)], &addr) == 1)
-            {
-                LogInfo("Got nameserver #%d: %s", mUpstreamDnsServerCount, &line.c_str()[sizeof(kNameserverItem)]);
-                mUpstreamDnsServerList[mUpstreamDnsServerCount] = addr;
-                mUpstreamDnsServerCount++;
-            }
+        const char *addressString = &line.c_str()[sizeof(kNameserverItem)];
+
+        if (inet_pton(AF_INET, addressString, &ip4Address) == 1)
+        {
+            otIp4ToIp4MappedIp6Address(&ip4Address, &ip6Address);
+            LogInfo("Got nameserver #%d: %s", mUpstreamDnsServerCount, addressString);
+            mUpstreamDnsServerList[mUpstreamDnsServerCount] = ip6Address;
+            mUpstreamDnsServerCount++;
+        }
+        else if (inet_pton(AF_INET6, addressString, &ip6Address) == 1)
+        {
+            LogInfo("Got nameserver #%d: %s", mUpstreamDnsServerCount, addressString);
+            mUpstreamDnsServerList[mUpstreamDnsServerCount] = ip6Address;
+            mUpstreamDnsServerCount++;
         }
```

Comment:
Can make it simpler? - avoid repeated code between `if` and `else` blocks?
```suggestion
        if (inet_pton(AF_INET, addressString, &ip4Address) == 1)
        {
            otIp4ToIp4MappedIp6Address(&ip4Address, &ip6Address);
        }
        else if (inet_pton(AF_INET6, addressString, &ip6Address) != 1)
        {
            continue;
        }

        LogInfo("Got nameserver #%u: %s", mUpstreamDnsServerCount, addressString);
        mUpstreamDnsServerList[mUpstreamDnsServerCount] = ip6Address;
        mUpstreamDnsServerCount++;
```